### PR TITLE
Jira PRO-308 Oui NotiTray Refactor

### PR DESCRIPTION
**Pull request by peterztan**
Refactored Notification tray, add storybook usages example and vanilla js basic example.

### CODE REVIEW

File: src/components/notification-tray/notification-tray.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+   */
+  @Event() public dismissall: EventEmitter
+  public dismissallHandler = () => {
+    this.dismissall.emit()
+  }
+
+  /**
+   *
+   */
+  @Event() public dismiss: EventEmitter
```
Comment (nnnnathann) [inline src/components/notification-tray/notification-tray.tsx]:
```suggestion
  @Event() public dismiss: EventEmitter<string>
```
This generic type can be better expressed by specifying "string" as the type parameter (custom element detail type), which tells the subscribing event handlers in typescript what type the event detail is, and can be used to validate the handler code

File: src/components/notification-tray/notification-tray.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  public dismissHandler = (el: HTMLElement) => {
+    this.dismiss.emit(el.dataset.notificationName)
+  }
+
+  @Listen("click")
+  public todoCompletedHandler(evt: UIEvent) {
+    evt.preventDefault()
+    evt.stopPropagation()
+    const target: HTMLElement = evt.target as HTMLElement
```
Comment (nnnnathann) [inline src/components/notification-tray/notification-tray.tsx]: No need to declare the type on both left hand of assignment and through coercion on the right hand side using `as`, typescript is smart enough to infer from the `as`

File: src/components/notification-tray/readme.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+      }
+    }
+
+    const drawer = document.querySelector("oui-notification-drawer")
+
+    const tray = document.querySelector("oui-notification-tray")
+    tray.count = notifications.length
+    tray.unread = notifications.length
+
+    const notificationsMapper = () => {
```
Comment (nnnnathann) [inline src/components/notification-tray/readme.md]: I think if we could, lets move this style of generation of the elements to a note in the readme (separate from an example), and have the usage example be just the markup

```html
<div style="width: 98%; margin: 4em auto; text-align: center;">
    <oui-tm-switch></oui-tm-switch>
    <oui-nav-bar>
      <div style="flex: 1;"></div>
      <oui-notification-tray>
        <oui-notification-drawer>
          <oui-notification-item name="Ubuntu Release" type="link" detail="Ubuntu 19.10 has been released" valence="success" />
          <oui-notification-item name="Apple Release" type="info" detail="OSX Catalina has been released" valence="fail" />        </oui-notification-drawer>
      </oui-notification-tray>
    </oui-nav-bar>
  </div>
```

**Comment by nnnnathann** [comment]
good work @peterztan , left some comments

File: src/components/notification-tray/notification-tray.tsx (hunk 2 above)
Comment (eevargas) [reply src/components/notification-tray/notification-tray.tsx]:
```suggestion
    const target = evt.target as HTMLElement
```

File: src/components/nav-bar/nav-bar.css [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+background-color: rgb(40, 51, 68);
+position: relative;
+top: 0;
+width: inherit;
+box-shadow: 2px 2px 10px rgba(0, 0, 0, 0.2);
+border-radius: 50px;
+/* overflow: visible; */
+}
+
+oui-nav-bar {
```
[the remark this answers is not in the document]
Comment (peterztan) [reply src/components/nav-bar/nav-bar.css]: I think it should be fine as long as the browser supports custom elements, but I haven't found much in regards to that specific issue.

File: .storybook/stories/notification-tray.stories.jsx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+                    />
+                  )
+                })
+              }
+            </oui-notification-drawer>
+          </oui-notification-tray>
+        </oui-nav-bar>
+      </div>
+
+      {/* <div style={ { width: "98%", margin: "4em auto", textAlign: "center" } }>
```
[the remark this answers is not in the document]
Comment (peterztan) [reply .storybook/stories/notification-tray.stories.jsx]: done :)

File: src/components/notification-drawer/notification-drawer.tsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  @Prop() public disabled: boolean = false
+
+  public render() {
+    return (
+      <Host class="oui-notification-drawer__statuses">
+        <button
+          disabled={this.disabled}
+          class="oui-notification-drawer__clear"
+          onClick={(evt: UIEvent) => this.onClickHandler(evt, this.disabled)}
+        >Clear All</button>
```
[the remark this answers is not in the document]
Comment (peterztan) [reply src/components/notification-drawer/notification-drawer.tsx]: done :)

File: src/components/notification-tray/notification-tray.tsx (hunk 1 above)
Comment (peterztan) [reply src/components/notification-tray/notification-tray.tsx]: done :)

File: src/components/notification-tray/notification-tray.tsx (hunk 2 above)
Comment (peterztan) [reply src/components/notification-tray/notification-tray.tsx]: done :)

**Pull request merged**